[ADD] mrp: dismantling incoming products for parts (#19296)
odoo/documentation: content/applications/inventory_and_mrp/manufacturing/workflows/dismantle_products.rst

split the item into components and add those parts to inventory.

Creating by-products ensures that the components for each product can be entered into inventory,
complete with serial numbers (if they are used).

Typically, the process is as follows:

#. :ref:`Create a bill of materials for the product to dismantle
   <manufacturing/dismantle_products/create-bom>`.
#. Create a manufacturing order (MO) for the product.
#. :ref:`Break down the product into its by-product components
   <manufacturing/dismantle_products/dismantle>` and enter any tracking numbers (like lots or serial
   numbers) for the component parts.
#. :ref:`Enter parts into inventory <manufacturing/dismantle_products/inventory>` by clicking
   :guilabel:`Produce All` on the manufacturing order.

.. _manufacturing/dismantle_products/config:

Enable settings
===============

To set up Odoo to dismantle products into by-products, navigate to :menuselection:`Manufacturing app
--> Configuration --> Settings`. In the *Operations* section, select the :guilabel:`By-Products`
checkbox.

.. image:: dismantle_products/by-products-setting.png

>    :alt: Enable By-Products in the Manufacturing configuration settings.

Reviewer: just pointing out: I like that this page is self-contained and a user could follow this end-to-end and you know how they love screenshots!

Pointing out the Byproducts checkbox and this screenshot also appears in both `byproducts.rst` (which this page links via :doc:) and `bill_configuration.rst` (which it links via the by-products :ref:). Again, i like that this page is self-contained. just flagging this in case it wasn't deliberate! :D

.. _manufacturing/dismantle_products/create-bom:

Create bill of materials
========================

First, create a new |BoM| for the product that will be dismantled.

Set up a :ref:`new BoM <manufacturing/basic_setup/bom-setup>`. Be sure to name the
product in a way that differentiates it from the product being disassembled.

Add the product being disassembled to the |BoM| as a component. Doing this will ensure that the

> number of disassembled components in stock will decrease by `1` every time an |MO| is completed.

Reviewer: rewording suggestion! optional

```suggestion
on-hand quantity of the product being disassembled will decrease by `1` every time an |MO| is completed.
```

:ref:`Add by-products <manufacturing/bill_configuration/by-products>` to the |BoM| that define the
components that will be entered into inventory as a result of an |MO|.

.. image:: dismantle_products/bom-byproducts.png
   :alt: A Car bill of materials with parts listed as by-products.

If necessary, add work orders and operations to the |BoM| to track disassembly.

After the |BoM| is created, create an |MO| and confirm it.

.. seealso::
   - :doc:`../basic_setup/one_step_manufacturing`
   - :doc:`../basic_setup/two_step_manufacturing`
   - :doc:`../basic_setup/three_step_manufacturing`

.. _manufacturing/dismantle_products/dismantle:

Dismantle product
=================

After the |MO| is confirmed, the product can be broken down into its component parts.

Retrieve the parts, or by-products, from the product.

If the by-products are tracked by lot or serial number, open the *By-Products* tab of the |MO|.

> Click the :icon:`fa-list` :guilabel:`(Open Move)` icon to open the *Open: Move Byproduct* window.

Reviewer 2: The MO needs to be confirmed before the Open Move icon will appear, right? Is that worth mentioning? I had to experiment to find it, but I know almost nothing about the Manufacturing app.

Author: So in the previous section, as well as a few lines above, I mention confirming the MO. That should be when the 'Open Move' icon appears.

===========================
Dismantle incoming products
===========================

Some companies may take in products from external sources and retain components of those products
for future use. For example, a car junkyard may take in cars that are no longer functional and break
them down into their components, selling the functional parts to mechanics, who can make use of
those parts.

Unlike :doc:`unbuild orders <unbuild_orders>`, incoming products that were not manufactured at the
company that is dismantling them can be broken down into their components using a manufacturing

> order with a bill of materials that creates component :doc:`by-products <byproducts>`.

Author: "While unbuild orders apply to products you manufactured, purchased products being disassembled rely on BoM by-products to split the item into components and add those parts to inventory."

Creating by-products ensures that the components for each product can be entered into inventory,
complete with serial numbers (if they are used).

Typically, the process is as follows:

#. :ref:`Create a bill of materials for the product to dismantle
   <manufacturing/dismantle_products/create-bom>`.
#. Create a manufacturing order for the product.
#. :ref:`Break down the product into its by-product components
   <manufacturing/dismantle_products/dismantle>` and enter any tracking numbers (like lots or serial
   numbers) for the component parts.
#. :ref:`Enter parts into inventory <manufacturing/dismantle_products/inventory>` by clicking
   :guilabel:`Produce All` on the manufacturing order.

> 

Author: Need a section about enabling by-products in settings.

.. _manufacturing/dismantle_products/create-bom:

Create bill of materials
========================

First, create a new bill of materials for the product that will be dismantled.

Set up a :ref:`new bill of materials <manufacturing/basic_setup/bom-setup>`.

> Because the product will be dismantled, no components are necessary. :ref:`Add by-products

Author: Correction from caolu: The product to be dismantled *is* the component -- this will ensure the stock of that item is reduced by 1 unit when the MO is completed, and all by-products are registered as necessary.

<manufacturing/bill_configuration/by-products>` to the bill of materials that define the components
that will be entered into inventory as a result of a manufacturing order.

.. image:: dismantle_products/bom-byproducts.png
   :alt: A Car bill of materials with parts listed as by-products.

> 

Author: Do we need to add information around creating an operation for the breakdown process?
